File: MyDailyBeat/MyDailyBeat/EVCMyHealthViewController.m
//
//  EVCMyHealthViewController.m
//  MyDailyBeat
//
//  Created by Virinchi Balabhadrapatruni on 5/30/16.
//  Copyright © 2016 eVerveCorp. All rights reserved.
//

#import "EVCMyHealthViewController.h"
#import "EVCCommonMethods.h"

@interface EVCMyHealthViewController ()

@end

@implementation EVCMyHealthViewController

- (id) initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil andPrescriptionProvider: (PrescripProviderInfo *) provider andHealthPortal: (HealthInfo *) portal {
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        self.provider = provider;
        self.portal = portal;
    }
    return self;
}

- (void)viewDidAppear:(BOOL)animated {
    [super viewDidAppear:animated];
    NSData *data1 = [[NSUserDefaults standardUserDefaults] objectForKey:@"myHealthPortal"];
    HealthInfo * temp1 = [NSKeyedUnarchiver unarchiveObjectWithData:data1];
    if (temp1 != nil) {
        self.portal = temp1;
    }
    NSData *data2 = [[NSUserDefaults standardUserDefaults] objectForKey:@"myPrescripProvider"];
    PrescripProviderInfo * temp2 = [NSKeyedUnarchiver unarchiveObjectWithData:data2];
    if (temp2 != nil) {
        self.provider = temp2;
    }
    
    [self setup];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (void) setup {
    if (self.provider != nil && self.portal != nil) {
        if (![self.portal.logoURL isEqualToString:@""]) {
            dispatch_queue_t queue = dispatch_queue_create("dispatch_queue_t_dialog", NULL);
            dispatch_async(queue, ^{
                dispatch_async(dispatch_get_main_queue(), ^{
                    [self.view makeToastActivity];
                });
                NSURL *imgurl = [NSURL URLWithString:self.portal.logoURL];
                NSData *data = [[RestAPI getInstance] fetchImageAtRemoteURL:imgurl];
                UIImage *img = [[UIImage alloc] initWithData:data];
                dispatch_async(dispatch_get_main_queue(), ^{
                    [self.view hideToastActivity];
                    [healthPortalLogoView setImage:[EVCCommonMethods imageWithImage:img scaledToSize:CGSizeMake(304, 128)]];
                });
            });
        } else {
            UIImage *img = [UIImage imageNamed:@"404-logo"];
            [healthPortalLogoView setImage:[EVCCommonMethods imageWithImage:img scaledToSize:CGSizeMake(304, 128)]];
        }
        
        if (![self.provider.logoURL isEqualToString:@""]) {
            dispatch_queue_t queue = dispatch_queue_create("dispatch_queue_t_dialog", NULL);
            dispatch_async(queue, ^{
                dispatch_async(dispatch_get_main_queue(), ^{
                    [self.view makeToastActivity];
                });
                NSURL *imgurl = [NSURL URLWithString:self.provider.logoURL];
                NSData *data = [[RestAPI getInstance] fetchImageAtRemoteURL:imgurl];
                UIImage *img = [[UIImage alloc] initWithData:data];
                dispatch_async(dispatch_get_main_queue(), ^{
                    [self.view hideToastActivity];
                    [prescripProviderLogoView setImage:[EVCCommonMethods imageWithImage:img scaledToSize:CGSizeMake(304, 128)]];
                });
            });
        } else {
            UIImage *img = [UIImage imageNamed:@"404-logo"];
            [prescripProviderLogoView setImage:[EVCCommonMethods imageWithImage:img scaledToSize:CGSizeMake(304, 128)]];
        }
        
        [prescripLabel setText:self.provider.URL];
        [portalLabel setText:self.portal.URL];
    } else {
        [prescripLabel setText:@"Prescription Provider not set"];
        [portalLabel setText:@"Health Portal not set"];
    }
}

- (IBAction)goToProvider:(id)sender {
    if (self.provider != nil) {
        NSString *trueU = [@"http://www." stringByAppendingString:self.provider.URL];
        [self openURLinBrowser:trueU];
    }
}

- (IBAction)goToPortal:(id)sender {
    if (self.portal != nil) {
        [self openURLinBrowser:self.portal.URL];
    }
}

- (void) openURLinBrowser: (NSString *) url {
    NSString *fullURL = [NSString stringWithFormat:@"%@", url];
    [[UIApplication sharedApplication] openURL:[NSURL URLWithString:fullURL]];
    
}

@end

File: MDB_Swift/MyDailyBeat/MyDailyBeat/src/EVCMyHealthViewController.swift

//
//  EVCMyHealthViewController.swift
//  MyDailyBeat
//
//  Created by Virinchi Balabhadrapatruni on 5/30/16.
//  Copyright © 2016 eVerveCorp. All rights reserved.
//
import UIKit
import Toast_Swift
import API
class EVCMyHealthViewController: UIViewController {

    @IBOutlet var prescripProviderLogoView: UIImageView!
    @IBOutlet var healthPortalLogoView: UIImageView!
    @IBOutlet var prescripLabel: UILabel!
    @IBOutlet var portalLabel: UILabel!
    @IBOutlet var prescripTap: UITapGestureRecognizer!
    @IBOutlet var healthTap: UITapGestureRecognizer!
    var provider: PrescripProviderInfo?
    var portal: HealthInfo?

    @IBAction func go(toProvider sender: Any) {
        if let provider = self.provider {
            let trueU: String = "http://www." + provider.url
            self.openURLinBrowser(trueU)
        }
    }

    @IBAction func go(toPortal sender: Any) {
        if let portal = self.portal {
            self.openURLinBrowser(portal.url)
        }
    }


    override func viewDidAppear(_ animated: Bool) {
        super.viewDidAppear(animated)
        let data1: Data? = UserDefaults.standard.object(forKey: "myHealthPortal") as? Data
        if let portal = data1, let temp1 = NSKeyedUnarchiver.unarchiveObject(with: portal) as? HealthInfo {
            self.portal = temp1
        }
        let data2: Data? = UserDefaults.standard.object(forKey: "myPrescripProvider") as? Data
        if let provider = data2, let temp2 = NSKeyedUnarchiver.unarchiveObject(with: provider) as? PrescripProviderInfo {
            self.provider = temp2
        }
        self.setup()
    }

    

    func setup() {
        if let provider = self.provider {
            if provider.logoURL != "" {
                
                DispatchQueue.global().async(execute: {() -> Void in
                    DispatchQueue.main.async(execute: {() -> Void in
                        self.view.makeToastActivity(ToastPosition.center)
                    })
                    if let imgurl = URL(string: provider.logoURL) {
                        let data = RestAPI.getInstance().fetchImage(atRemoteURL: imgurl)
                        let img = UIImage(data: data)
                        DispatchQueue.main.async(execute: {() -> Void in
                            self.view.hideToastActivity()
                            self.prescripProviderLogoView.image = EVCCommonMethods.image(with: img!, scaledTo: CGSize(width: CGFloat(304), height: CGFloat(128)))
                        })
                    } else {
                        DispatchQueue.main.async(execute: {() -> Void in
                            self.view.hideToastActivity()
                        })
                    }
                    
                })
            }
            prescripLabel.text = provider.url
        } else {
            prescripLabel.text = "Prescription Provider not set"
        }
        
        if let portal = self.portal {
            if portal.logoURL != "" {
                
                DispatchQueue.global().async(execute: {() -> Void in
                    DispatchQueue.main.async(execute: {() -> Void in
                        self.view.makeToastActivity(ToastPosition.center)
                    })
                    let imgurl = URL(string: portal.logoURL)
                    let data: Data? = RestAPI.getInstance().fetchImage(atRemoteURL: imgurl!)
                    let img = UIImage(data: data!)
                    DispatchQueue.main.async(execute: {() -> Void in
                        self.view.hideToastActivity()
                        self.healthPortalLogoView.image = EVCCommonMethods.image(with: img!, scaledTo: CGSize(width: CGFloat(304), height: CGFloat(128)))
                    })
                })
            }
            
            portalLabel.text = portal.url
        } else {
            portalLabel.text = "Health Portal not set"
        }
    }

    func openURLinBrowser(_ url: String) {
        let fullURL: String
        if url.hasPrefix("http://") || url.hasPrefix("https://") {
            fullURL = "\(url)"
        } else {
            fullURL = "http://\(url)"
        }
        UIApplication.shared.open(URL(string: fullURL)!, options: [:], completionHandler: nil)
    }
}
